ruby-challenges.rb

```diff
@@ -3,26 +3,98 @@
 
 # --------------------1) Create a method that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
-beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
 
-letter_o = 'o'
+
+# beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
+
+# letter_o = 'o'
+
+# so my thinking is to find a method that will iterate through the array for a matching element
+# create a variable 
+# give it a function that takes in a variable 
+# returns a array of values that have the element
+
+drank = []
+beverages_array.each do |s|
+  if letter_o
+    return s 
+  end
+end
+
+
+# print frameworks.select {|i| beverages_array.find == {letter_o}
+
+# same
+
 # Expected output: ['coffee', 'soda water']
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+drank = []
+beverages_array_t.each do |s|
+  if letter_t
+    return s 
+  end
+end
+
+
 
 # -------------------2) Create a method that takes in an array of numbers and returns the sum of the numbers. Use the test variables provided.
 
-nums_array1 = [42, 7, 27]
-# Expected output: 76
+# nums_array1 = [42, 7, 27]
+# # Expected output: 76
 
-nums_array2 = [25, 17, 47, 11]
+# create a variable 
+# give it a function that takes in a variable and adds it to the last value
+# use the .each method to iterate through the array
+# returns a array of values that have the element
+
+sum_of_each.each do |s|
+    s.each { |a| sum+=a }
+    end
+end
+
```

Comment:
I would be curious to know what your output is here. I am not sure what `sum_of_each` is referencing.